### DISCUSSION TITLE
Trilium & NGINX Proxy Manager

### DISCUSSION THREAD

**Discussion by Rgamer84** [TriliumNext/Trilium#3531] [Q&A] [answered]
When creating/deleting notes, the note does not automatically appear or disappear and requires a refresh to update the page every time.  I also noticed the same issue when just trying to hide or open the panel.  This issue is NOT an issue with Trilium but instead something with NGINX Proxy Manager.  If I remove the proxy manager from the equation everything works as it should.  Also, I did not have this issue with SWAG but recently decided I wanted something that was a little easier to manage so it's not even really specific to nginx itself it seems.

Other things to note:  I do use Authentik however I've removed all config for that to rule that out.  I've also attempted to disable Cache Assets, Websockets Support, Block Common Exploits as well as removed all SSL related items to no avail.  I have attempted both Firefox and Chrome as well as using incognito to rule out a cookie issue however the issue persists.

My question is, has anyone else ran into this or have any ideas of what might be causing me to have to refresh?  If you are using NGINX Proxy Manager and not having these issues, can you let me know what you have set so I can attempt to see if it helps me out.

**Comment by sigaloid** [comment, marked as answer]
Are you using the [recommended configuration found here?](https://github.com/zadam/trilium/wiki/Nginx-proxy-setup) If so, perhaps press F12 and check for failed network requests/errors logged.

**Reply by Rgamer84** [reply]
I had looked at that before but many of the items there according to my instance were deprecated or redundant in the config and threw all kinds of errors.  The part here that differentiates the proxy manager from the basic nginx config is that the proxy manager is writing out the config based off of radio button toggles so it's sorting out what is/isn't in the config trying to compare.  I did however from what you sent do some trial and error and found that adding the following lines listed in the recommended config corrected the behavior I was seeing:

```
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "upgrade";
```

I'm not entirely sure why those two items specifically would resolve the behavior that I was seeing but it resolved it. Thanks for the pointer!

**Reply by sigaloid** [reply]
Yep, as zadam mentioned, that's related to the websocket connection that must be made. Glad you got it working :smile:

**Comment by zadam** [comment]
Hi, this happens when websocket communicatin is not going through - usually proxy server blocks it.
